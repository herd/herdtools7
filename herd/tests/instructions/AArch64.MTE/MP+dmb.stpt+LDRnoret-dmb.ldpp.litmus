AArch64 MP+dmb.stpt+LDRnoret-dmb.ldpp

{
 0:X1=x; 0:X2=y:red; 0:X3=y:green;
 1:X1=x; 1:X3=y:red;
}
 P0          | P1               ;
 MOV W0,#1   | L0: LDR WZR,[X3] ;
 STR W0,[X1] | DMB LD           ;
 DMB ST      |                  ;
 STG X2,[X3] | LDR W0,[X1]      ;

exists (not (fault(P1:L0,y)) /\ 1:X0=0)

