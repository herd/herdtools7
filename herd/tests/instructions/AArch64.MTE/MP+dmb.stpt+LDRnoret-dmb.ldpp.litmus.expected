Test MP+dmb.stpt+LDRnoret-dmb.ldpp Allowed
States 2
1:X0=0; Fault(P1:L0,y:red,TagCheck);
1:X0=1;  ~Fault(P1:L0,y);
No
Witnesses
Positive: 0 Negative: 2
Condition exists (not (fault(P1:L0,y)) /\ 1:X0=0)
Observation MP+dmb.stpt+LDRnoret-dmb.ldpp Never 0 2
Hash=d44dceb7810802a96e50c4617016b737

